### ISSUE TITLE
ValueError: Only float type quantization is supported. Weights onnx::Concat_1052 is INT64. 

### ISSUE THREAD

**Issue by jinfagang** [microsoft/onnxruntime#11313]
```
ValueError: Only float type quantization is supported. Weights onnx::Concat_1052 is INT64. 

```

Got this when quantize a valid model:

![image](https://user-images.githubusercontent.com/21303438/164654464-347933d6-6b40-494e-aa57-e436ec3b3c44.png)

**Comment by yufenglee** [comment]
Which version are you using? I cannot repro it with latest 1.11 with script like below: 

```
import numpy as np
from onnx import helper, TensorProto, numpy_helper, save
from onnxruntime.quantization import quantize_static, QuantFormat, CalibrationDataReader

class TestDataFeeds(CalibrationDataReader):
    def __init__(self, data_feeds):
        '''
        parameter data_feeds: list of input feed, each input feed is diction of {input_name: np_array}
        '''
        self.data_feeds = data_feeds
        self.iter_next = iter(self.data_feeds)

    def get_next(self):
        return next(self.iter_next, None)

    def rewind(self):
        self.iter_next = iter(self.data_feeds)


def InputFeedsNegOneZeroOne(n, name2shape, is_input_int64):
    '''
    randomize n feed according to shape, its values are from -1, 0, and 1
    '''
    input_data_list = []
    for i in range(n):
        inputs = {}
        for name, shape in name2shape.items():
            inputs.update({name: np.random.randint(-1, 2, shape).astype(np.int64 if is_input_int64 else np.float32)})
        input_data_list.extend([inputs])
    dr = TestDataFeeds(input_data_list)
    return dr

class TestONNXModel:
    def construct_model(self, model_path, is_input_int64):
        input = helper.make_tensor_value_info('input', TensorProto.INT64 if is_input_int64 else TensorProto.FLOAT, [1, 3, 15, 15])
        output = helper.make_tensor_value_info('output', TensorProto.INT64 if is_input_int64 else TensorProto.FLOAT, [1, 13, 15, 15])

        # Conv1 output [1, 2, 13, 13]
        concat_weight_0 = numpy_helper.from_array(
            np.random.randint(-100, 200, [1, 5, 15, 15]).astype(np.int64 if is_input_int64 else np.float32), name='concat_weight_0')

        concat_weight_1 = numpy_helper.from_array(
            np.random.randint(-100, 200, [1, 5, 15, 15]).astype(np.int64 if is_input_int64 else np.float32), name='concat_weight_1')

        initializers = [concat_weight_0, concat_weight_1]

        concat_node = helper.make_node('Concat', ['input', 'concat_weight_0', 'concat_weight_1'], [
                                            'output'], name='concat_node', axis=1)

        graph = helper.make_graph([concat_node],
                                  'qlinear_concat_op_test', [input], [output], initializer=initializers)
        model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 13)])
        save(model, model_path)

    def test_quantize_concat(self, is_input_int64):
        np.random.seed(1)

        suffix = '.int64.onnx' if is_input_int64 else '.float32.onnx'
        model_fp32_path = 'concat' + suffix
        model_int8_path = 'concat_quant' + suffix
        model_int8_qdq_path = 'concat_quant_qdq' + suffix

        self.construct_model(model_fp32_path, is_input_int64)

        # Verify QOperator mode
        data_reader = InputFeedsNegOneZeroOne(1, {'input': [1, 3, 15, 15]}, is_input_int64)
        quantize_static(model_fp32_path, model_int8_path, data_reader, quant_format=QuantFormat.QOperator)

        # Verify QDQ mode
        data_reader.rewind()
        quantize_static(model_fp32_path, model_int8_qdq_path, data_reader)


TestONNXModel().test_quantize_concat(True)
TestONNXModel().test_quantize_concat(False)
```

**Comment by NeonBohdan** [comment]
The same, error fixed by switching onnxruntime 1.10->1.11
